### DISCUSSION TITLE
Enable Segmentation in Ros2Camera plugin in webots_ros2

### DISCUSSION THREAD

**Discussion by hardik-a62** [cyberbotics/webots_ros2#763] [Q&A] [answered]
Does the webots_ros2_driver Ros2Camera supportes enabling and publishing segmentation results?

**Comment by ygoumaz** [comment, marked as answer]
You can enable camera recognition in the URDF file (`<device>` tag) and two topics will be published: `/topicName/recognitions` and `/topicName/recognitions/webots`. They respectively receive a list of the positions of detected objects on the image and a list of detailed information about each detected objects, including its pose. In case you are interested about the segmentation image, [the plugin](https://github.com/cyberbotics/webots_ros2/blob/master/webots_ros2_driver/src/plugins/static/Ros2Camera.cpp) must be enhanced, as it currently doesn't enable segmentation and doesn't publish the segmentation image. If you open a PR to implement it, we would be happy to review it.

You can get more information about camera segmentation in the [Reference Manual of Webots](https://cyberbotics.com/doc/reference/camera#wb_camera_recognition_enable_segmentation)
